### PR TITLE
Remove downcase of restrict_with_error translations

### PR DESCRIPTION
**Pull request by sirkosi**
### Summary

This reverts the downcase of defined translations for `restrict_with_error`. The downcase might apply for the english language, but in case sensitive languages this behaviour is incorrect.

However, removing the downcase, this change upcases previous spellings for undefined translations. 

Related open Issue: https://github.com/rails/rails/issues/21064

### Other Information

Another idea to fix this might be to make the `default` option inside `ActiveModel::Translation.human_attribute_name` work instead of overwriting it, although this could impact the usage in other places.

### CODE REVIEW

**Comment by p8** [comment]
Hi @sir-ko 👋,  thanks for contributing to Rails and tackling this old issue!

Looking at other occurences of downcase I was expecting to see more cases where this could apply.
Maybe?:
https://github.com/rails/rails/blob/f95c0b7e96eb36bc3efc0c5beffbb9e84ea664e4/railties/lib/rails/generators/erb/scaffold/templates/new.html.erb.tt#L1

I think if we're gonna change this we would need to make it configurable per locale.
I can't find any similar implementations for example adding an option to the `config/locales/en.yml`
What comes closest I guess is the inflections file, but I'm not sure we want to add something similiar for 2 error messages:
https://github.com/rails/rails/blob/main/railties/lib/rails/generators/rails/app/templates/config/initializers/inflections.rb.tt

**Comment by eileencodes** [comment]
We can't make this change without a deprecation. If an app upgrades to Rails 7 all of a sudden their error messages will be capitalized when before they weren't. This change could easily get missed without a deprecation and would then look incorrect on product pages.

**Comment by sir-ko** [comment]
@p8 thanks for your feedback! I'm not sure if this applies to the scaffolding templates, since the surrounding language is in plain english. Or could you explain a little more, not sure if I can follow.

@eileencodes thank you! Good point, you are right! The current behaviour has to stay the same.

I was doing some refinement (9d6b8c2), which introduces a new method that respects the default parameter. Please let me know what you think.
